src/app/shared/components/product/product-add-to-basket/product-add-to-basket.component.ts

```diff
@@ -83,4 +83,13 @@ export class ProductAddToBasketComponent implements OnInit, OnDestroy {
   ngOnDestroy() {
     this.destroy$.next();
   }
+
+  hasAddToBasketButton() {
```

Comment:
Why here? Isn't this already decided by parent components?

Reply:
not really, ProductAddToBasketComponent is used in six other components and just "tile" and "row" has this condition in the ProductItemContainerConfiguration

Reply:
But then the logic is executed twice for product-rows and product-tiles, which is not what we want. Either we do it always from the outside or we do it from inside the component.

It seems that in ISML it is done in the component itself. But then there would be no need to have configuration from the outside.

So we either do it always from the outside (then we'll need a helper method after all) or we only let the add-to-basket component decide when to show itself.

opinions? @shauke

Reply:
The issue whether to show the Add To Cart button or not has two aspects.

It should be configurable at parent components like the product tile, where we might want to use the same product tile component at several places with different rendering styles showing the add to cart button at some places and not at others. So this needs to be for some parent components a display configuration (e.g. for the product tile and product row, but not for product detail or product compare). This is the case in ISML too (see ProductTile.isml).

On the other hand there needs to be logic at the Add To Cart component itself too, that determines whether the add to cart button can be show or not. This logic should be independent from the parents configuration to show the add to cart button. This logic is only based on product properties that are needed to show an add to cart button or not. This is the same in ISML too (see AddToCartInc.isml). And it should be independent from the parent so the Add To Cart component can be used in several places without the need to implement the same logic whether to display it or not in several parent components (compare listing, product tile, product row, product detail, ...).

Reply:
So as an example: The check for availability (inStock, ...) should be made within the component and disabling the button for master products should be done from the outside?